packages/cardpay-sdk/sdk/prepaid-card/base.ts

```diff
@@ -259,13 +286,15 @@ export default class PrepaidCard {
       );
     }
 
+    let txnHash = gnosisResult.ethereumTx.txHash;
     if (typeof onTxnHash === 'function') {
-      await onTxnHash(gnosisResult.ethereumTx.txHash);
+      await onTxnHash(txnHash);
     }
 
-    return await waitUntilTransactionMined(this.layer2Web3, gnosisResult.ethereumTx.txHash);
+    return await waitUntilTransactionMined(this.layer2Web3, txnHash);
   }
 
+  // TODO: add overload for passing txnHash
```

Comment:
For this one we are removing the airdrop of CARD.CPXD, so the additional txn's after the split will be deleted. So I think this overload will look like (note the removed `onGasLoaded`):
```ts
async split(
    txnHash: string
  ): Promise<{ prepaidCards: PrepaidCardSafe[]; txReceipt: TransactionReceipt }>;
async split(
    prepaidCardAddress: string,
    faceValues: number[],
    customizationDID: string | undefined,
    onPrepaidCardsCreated?: (prepaidCards: PrepaidCardSafe[], txnHash: string) => unknown,
    txnOptions?: TransactionOptions,
    contractOptions?: ContractOptions
  ): Promise<{ prepaidCards: PrepaidCardSafe[]; txReceipt: TransactionReceipt }>;
async split(
    prepaidCardAddressOrTxnHash: string,
    faceValues: number[],
    customizationDID: string | undefined,
    onPrepaidCardsCreated?: (prepaidCards: PrepaidCardSafe[], txnHash: string) => unknown,
    txnOptions?: TransactionOptions,
    contractOptions?: ContractOptions
  ): Promise<{ prepaidCards: PrepaidCardSafe[]; txReceipt: TransactionReceipt }> {
.
.
}
``` 
The prepaid card addresses can be derived from the `txnHash` when solely the `txnHash` is passed in. You can find the code below for doing this, but it looks like this: 
```ts
let prepaidCardAddresses = await this.getPrepaidCardsFromTxn(gnosisResult.ethereumTx.txHash);
```

Reply:
> For this one we are removing the airdrop of CARD.CPXD, so the additional txn's after the split will be deleted. So I think this overload will look like (note the removed onGasLoaded and onPrepaidCardsCreated):

Can I go ahead and remove those arguments and the associated code in this PR or is there another one pending?

Reply:
yes, I think that is fine. The work to do this is on my list. so if you get it done first, that's one less thing for me to worry about :-)